source/m2mconnectionhandlerpimpl.cpp

```diff
@@ -154,6 +154,7 @@ M2MConnectionHandlerPimpl::~M2MConnectionHandlerPimpl()
     stop_listening();
 
     close_socket();
+    pal_destroy();
```

Comment:
To play it safe, I would suggest moving the destructor to line after "delete _security_impl;". Just in case that part has not called the the pal_init() itself and relies on functioning PAL on its destructor.